### DISCUSSION TITLE
Support for the "on" and "off" buttons on Hub 2?

### DISCUSSION THREAD

**Discussion by DistractedPhysicist** [hsakoh/switchbot-mqtt#28] [Q&A] [answered]
Hi, thank you very much for creating this add-on, it's been working great for me. However, I didn't find a way to configure the "on" and "off" buttons of Hub 2 to trigger some HA automation for instance (the reason why I want to do this is because SwitchBot scenes do not allow conditions presently). Am I missing something or is this currently not supported, and in the latter case, will there potentially be support for it in the future? Thanks! Best regards, Ville

**Comment by hsakoh** [comment, marked as answer]
Which of the following do you want to achieve?

1. Detect when the On/Off button of Hub 2 is pressed as an event in Home Assistant.
2. Press the On/Off button of Hub 2 from Home Assistant.

For case 1, Hub 2's Webhook only notifies changes in temperature and humidity.
https://github.com/OpenWonderLabs/SwitchBotAPI?tab=readme-ov-file#hub-2-1

For case 2, Hub 2 does not have DeviceControlCommands implemented.
https://github.com/OpenWonderLabs/SwitchBotAPI?tab=readme-ov-file#send-device-control-commands
While it's not possible to press the On/Off button, this add-on does support executing scenes.

(For case 1, although it's quite an unconventional approach... 
You could set up a scene that is executed when the On/Off button is pressed, which then operates a device that sends a Webhook notification (e.g., PlugMini). This Webhook can then be received and detected by Home Assistant.)

It would be better to raise an issue in the SwitchBotAPI repository for adding Webhook support or control commands.
https://github.com/OpenWonderLabs/SwitchBotAPI

**Reply by DistractedPhysicist** [reply]
Thank you for your detailed reply! I am currently interested in case 1. For the unconventional approach you suggested, I am a complete newbie so I might be misunderstanding something, but I think that since all the actions the Hub 2 buttons trigger are actions also triggered by other things throughout the day, HA would not be able to reliably identify the specific Webhook originating from the button press. I also raised an issue about Webhook support for Hub 2 buttons as you suggested: https://github.com/OpenWonderLabs/SwitchBotAPI/issues/324 (hopefully I was able to explain it well enough!) Thank you again for your super helpful reply!

**Comment by zalmael** [comment]
Just for the record, adding the hub2 with matter allows to catch in HA when on/off are pressed

**Reply by DistractedPhysicist** [reply]
Hi @zalmael , and thank you for the suggestion! I tried this according to the instructions here: https://www.creatingsmarthome.com/index.php/2024/04/25/matter-with-switchbot-hub-2-and-home-assistant/
But after the final step, it loads for a while, and then says "Unable to Add Accessory" (this is different from the article, where they got "Something went wrong", and it still worked; in my case no Matter device gets added) Am I doing something wrong? I have no clue how I could begin to figure out whether the issue is with the Hub 2 or HA, or the companion app... Or could this somehow be a compatibility issue with the SwitchBot-Mqtt add-on, @hsakoh ? Also, if there is any additional information that would help the diagnosis please let me know (I tried checking the companion app log but it only seems to contain location info)! I really appreciate all your help, thank you again!

![IMG_0473](https://github.com/user-attachments/assets/bf77ea0d-3ec4-4598-a527-8d0dc8a17fd5)
![IMG_0474](https://github.com/user-attachments/assets/fb487b95-463b-4f63-9b32-540305e7cbd8)
![IMG_0475](https://github.com/user-attachments/assets/3a97d47a-a345-44ba-85fa-adfd7924bd0f)

**Reply by hsakoh** [reply]
> Or could this somehow be a compatibility issue with the SwitchBot-Mqtt add-on,

The add-on and Matter are completely unrelated and do not interfere with each other. Therefore, there are no issues with compatibility or interoperability. Additionally, at least in my environment, I am able to add devices with Matter.

**Reply by DistractedPhysicist** [reply]
Thank you for your quick reply @hsakoh , that's what I thought. I finally figured out what was wrong: my iPhone was connected to the 5 GHz Wi-Fi instead of the 2.4 GHz one that the Hub 2 uses. Everything worked beautifully once I switched my iPhone Wi-Fi. I was also able to add the ON and OFF buttons to HA and trigger automations with them, which is exactly what I wanted, thanks again for pointing this out @zalmael :)

**Reply by zalmael** [reply]
Glad I helped ! :)
